File: 31_10_2022/py/#2_cdck.py
max_ = 0
start, stop = [int(i) for i in open('./cdck.inp').readline().split()]
for j in range(start, stop+1):
	count = 1
	i = j
	while i != 1:
		if not i % 2:
			i //= 2
		else:
			i = 3*i + 1
		count += 1
	max_ = max(count, max_)
print(max_, file=open('./cdck.out', 'w+'))

File: 31_10_2022/cpp/#2_cdck.cpp
#include <iostream>

int main() {
	freopen("./cdck.inp", "r", stdin);
	freopen("./cdck.out", "w+", stdout);

	int start, stop;
	std::cin >> start >> stop;

	int max = 0;
	for (int i = start; i <= stop; i++) {
		int j = i;
		int count = 1;
		while (j != 1) {
			if (j % 2 == 0) {
				j /= 2;
			} else {
				j = 3*j + 1;
			}
			count++;
		}
		max = (count>max)?count:max;
	}

	std::cout << max;
}